lib/src/value/calculation.dart

```diff
@@ -158,6 +382,248 @@ class SassCalculation extends Value {
+    _verifyLength([y, if (x != null) x], 2);
+    y = _simplify(y);
+    x = x.andThen(_simplify);
+    var args = [y, if (x != null) x];
+    if (y is! SassNumber ||
+        x is! SassNumber ||
+        y.hasUnit('%') ||
+        x.hasUnit('%')) {
+      return SassCalculation._("atan2", args);
+    }
+    _verifyCompatibleNumbers(args);
+    return number_lib.atan2(y, x);
+  }
+
+  /// Creates a `rem()` calculation with the given [dividend] and [modulus].
+  ///
+  /// Each argument must be either a [SassNumber], a [SassCalculation], an
+  /// unquoted [SassString], a [CalculationOperation], or a
+  /// [CalculationInterpolation].
+  ///
+  /// This automatically simplifies the calculation, so it may return a
+  /// [SassNumber] rather than a [SassCalculation]. It throws an exception if it
+  /// can determine that the calculation will definitely produce invalid CSS.
+  ///
+  /// This may be passed fewer than two arguments, but only if one of the
+  /// arguments is an unquoted `var()` string.
+  static Value rem(Object dividend, Object? modulus) {
+    dividend = _simplify(dividend);
+    modulus = modulus.andThen(_simplify);
+    var args = [dividend, if (modulus != null) modulus];
+    _verifyLength(args, 2);
+    if (dividend is! SassNumber ||
+        modulus is! SassNumber ||
+        dividend.hasUnit('%') ||
+        modulus.hasUnit('%')) {
+      return SassCalculation._("rem", args);
+    }
+    _verifyCompatibleNumbers(args);
+    if (modulus.value.isInfinite) return dividend;
+    var modulo = dividend.modulo(modulus);
+    return (modulo is SassNumber && modulo.value == 0)
+        ? SassNumber(-0.0)
+        : modulo.minus(dividend);
+  }
+
+  /// Creates a `mod()` calculation with the given [dividend] and [modulus].
+  ///
+  /// Each argument must be either a [SassNumber], a [SassCalculation], an
+  /// unquoted [SassString], a [CalculationOperation], or a
+  /// [CalculationInterpolation].
+  ///
+  /// This automatically simplifies the calculation, so it may return a
+  /// [SassNumber] rather than a [SassCalculation]. It throws an exception if it
+  /// can determine that the calculation will definitely produce invalid CSS.
+  ///
+  /// This may be passed fewer than two arguments, but only if one of the
+  /// arguments is an unquoted `var()` string.
+  static Value mod(Object dividend, Object? modulus) {
+    dividend = _simplify(dividend);
+    modulus = modulus.andThen(_simplify);
+    var args = [dividend, if (modulus != null) modulus];
+    _verifyLength(args, 2);
+    if (dividend is! SassNumber ||
+        modulus is! SassNumber ||
+        dividend.hasUnit('%') ||
+        modulus.hasUnit('%')) {
+      return SassCalculation._("mod", args);
+    }
+    _verifyCompatibleNumbers(args);
+    if (modulus.value.isInfinite &&
+        ((modulus.value < 0 && dividend.value >= 0) ||
+            modulus.value >= 0 && dividend.value < 0)) {
+      return SassNumber(double.nan).coerceToMatch(dividend);
+    }
```

Comment:
This shouldn't be necessary by the calc function spec (`dividend.modulo(modulus)` should already handle any special cases for `mod`, and we don't have any extra special cases like with `rem`)

Reply:
Although, this line was added in the spec `*If c2 is infinity and has a different sign than c1 (including oppositely-signed zero), return NaN with the same units as c1.`. So if I return `dividend.modulo(modulus)` directly, in this test case: `a {b: mod(10, -infinity)}` the result is `calc(-infinity)` instead of `calc(NaN)`

Reply:
Ah, right. In that case though, the change should be made to the `SassNumber` `modulo` method so that it applies to the `%` in addition to here.